### PR TITLE
JSRS Migration

### PR DESCRIPTION
**Pull request by fdsgusmao**

### CODE REVIEW

**Comment by fdsgusmao** [comment]
### Goals Verified:

1. GetContent
    - start
    - tell me a fact about dogs
    - tell me a fact
    - tell me a cat joke
    - tell me a joke

**NO ERRORS:** The migration tool successfully migrated the code

**Review by kangawoo44 — APPROVED** [review]
LGTM

**Comment by fdsgusmao** [comment]
@kangawoo44 

- Warnings:0
- Errors:0
- Stage-1 Deprecations:0
- Stage-2 Deprecations:0

image

File: capsule.bxb [hunk 1]
```diff
@@ -1,8 +1,10 @@
 capsule {
   id (example.fact)
   version (1.0.0)
-  format(3)
-  runtime-version (2)
+  format (3)
+  runtime-version (2) {
```
Comment (kangawoo44) [inline capsule.bxb]: update to runtime 9

**Comment by fdsgusmao** [comment]
After updating runtime-version to 9 an issue with the dialogs appeared, they simply did not show with the view, so I made a conditional change to the view and it worked, not sure if it is the optimal solution but it worked, if you have a suggestion on how to do it better I'll be happy to implement it

**To solve the dialog issue I had to implement the conditions inside the view file instead of using different dialog files**
image

**Comment by kangawoo44** [comment]
take a look at the runtime flags here to see which runtime flag could have affected the dialog behavior (going from version 2 to 9).
https://corp.bixbydevelopers.com/dev/docs/dev-guide/developers/customizing-plan.effects#runtime-versions-and-runtime-flags

I'm guessing it's likely the [no-fallback-dialog-in-views](https://corp.bixbydevelopers.com/dev/docs/reference/type/capsule.runtime-version.overrides.no-fallback-dialog-in-views) flag. Can you run an experiment with this flag and find out if this is it? We should understand the why so we can write the most optimal solution for the issue encountered. After you find the flag responsible for this issue, let me know if you would keep the same solution you've written or if you would change it.

**Comment by fdsgusmao** [comment]
> I'm guessing it's likely the [no-fallback-dialog-in-views](https://corp.bixbydevelopers.com/dev/docs/reference/type/capsule.runtime-version.overrides.no-fallback-dialog-in-views) flag. Can you run an experiment with this flag and find out if this is it?

**That was it! This is before the flag:**

image


**And after the flag:**
image

**Comment by kangawoo44** [comment]
@fdsgusmao I was thinking we would avoid overriding the runtime flag but rather find a solution that works around it. What you previously had, for example, worked around the flag. But now that you have a better understanding of what was causing the issue, can you think about whether you want to keep your previous implementation or change something about it to present as an optimal solution?

**Comment by fdsgusmao** [comment]
@kangawoo44 
**To keep the dialogs showing without overriding the flag we need to define a message inside the view, from what I can tell there are 2 ways to do that**

**1. Manually invoking the dialog events,** if no result is found call `NoResult` dialog  else call `Result` dialog:
```JS
message {
    if (size(this) == 0) {
      template ("#{event(this, 'NoResult')}")
    } else {
      template ("#{event(this, 'Result')}")
    }
  }
```

2. The first solution I came up with:
```JS
message{
    if (exists(this.text)){
      template ("#{raw(this.text)}")
    }else{
      template ("Nothing Found")
    }
  }
```

### Conclusion
Taking in consideration that this is a sample capsule, in my opinion it's better to go with solution 2 because it's easier to understand and overall beginner friendly

**Comment by kangawoo44** [comment]
That sounds good to me. @abapayeViv what do you think?

**Comment by abapayeViv** [comment]
That makes sense to me too. The 2nd one is much more straightforward.

**Comment by fdsgusmao** [comment]
Great, I hard reseted the branch to the previous commit, now the 2nd solution is implemented

**Review by abapayeViv — APPROVED** [review]
👍

**Pull request merged**